### DISCUSSION TITLE
How to handle silent updates

### DISCUSSION THREAD

**Discussion by ltguillaume** [ScoopInstaller/Scoop#5657] [Ideas]
Some developers do "silent updates", where the version numbers stays the same, but the release is updated. As such, Scoop will fail to update the application, because the hash won't match up with the download anymore.

Some manifests use `autoupdate`->`architecture`->`64bit`->`hash`->`url` to define the place for the hashes. This will still be updated to the proper hash.

So to partly resolve this issue, I think it would be a good idea to let `checkver.ps1` check this file for changes, even if the version hasn't been updated.

**Comment by HUMORCE** [comment]
`checkhashes.ps1`: Compare hash of file with hash provided by manifest.

`checkver.ps1`: Execute with param `-f|-ForceUpdate` to updating manifest even no new version detected.

---

In the case you mentioned or no version, another dynamic value should be considered to appended/used as version of manifest.

e.g.: `app 1.1-f8725a6`, `app 1.1-230923`, `sysinternals 2023.7.26`

**Reply by ltguillaume** [reply]
> Compare hash of file with hash provided by manifest.

What I'd like to accomplish is a bit different, though: without downloading the entire file, I'd like to compare the hash provided by the manifest with the hash provided via the hash url.

**Reply by HUMORCE** [reply]
`checkver.ps1 <app> -ForceUpdate` will not download file when `autoupdate.hash` is defined and available.

---

For case your provided now, I will do like this:

```
version: 1.0-d5d348
checkver:
  script:
    $page = iwr <page_url>
    $version = <matched_version>
    $hash_min = <matched_hash>.substring(0,6).tolower()
    write-output "$version-$hash_min"
```

**Reply by ltguillaume** [reply]
That looks like a good solution, but I'm not sure how to implement this in the manifest. It would really be great if I could force this behavior from within a manifest file.

**Reply by HUMORCE** [reply]
I think I explained it enough detail, if not, try searching more similar cases in main/extras bucket, if you are using vscode, then: `checkver.*\n.*script":`(regex)

`extras/sysinternals` is a good example for this.
